src/msg/Connection.h

```diff
@@ -126,10 +126,12 @@ struct Connection : public RefCountedObject {
    *
    * @param m The Message to send. The Messenger consumes a single reference
    * when you pass it in.
-   *
+   * @param sent Callback called when the message is sent.
    * @return 0 on success, or -errno on failure.
    */
-  virtual int send_message(Message *m) = 0;
+  virtual int send_message(Message *m, std::function<void(Message*)> sent=nullptr) = 0;
```

Comment:
I'm not really enthusiastic about this API. How about setting a callback function for the Connection like:

`void set_message_sent_callback(std::function<void(const MessageRef& m)>);`

It should be up to the caller to associate context with each MessageRef.

(We should also avoid adding new methods that use raw Message pointers.)